sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/instrumentation/InstrumentationUtils.java

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.implementation.instrumentation;
+
+import java.net.URI;
+
+/**
+ * Utility class for instrumentation.
+ */
+public final class InstrumentationUtils {
+
+    /**
+     * Does the best effort to capture the server port with minimum perf overhead.
+     * If port is not set, we check scheme for "http" and "https" (case-sensitive).
+     * If scheme is not one of those, returns -1.
+     *
+     * @param uri request URI
+     * @return server port
+     */
+    public static int getServerPort(URI uri) {
+        int port = uri.getPort();
+        if (port == -1) {
+            switch (uri.getScheme()) {
```

Comment:
Should we add support for more schemes than just HTTP and HTTPS to loosen the tie to HTTP?

Reply:
maybe in the future, we can always extend the list.